docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,190 @@
+  been deleted then a new `SubscriberList` record will be generated.
+
+* **Unused SubscriberList**. Remove lists which have never had any
+  subscriptions and are over seven days old. A scenario of when this can happen
+  is when a user quits their signup journey midway through. Seven days was the
+  chosen retention period to allow users to complete their signup journey as
+  they have a [seven day limit] to confirm their subscription. As mentioned in
+  the context section, unused subscriber lists pollute our analytics and so we
+  should remove them regularly.
+
+* **Subscriber**. Remove subscribers once their `created_at` is over a year old
+  and they don't have any subscriptions (active or ended). Subscribers should
+  only be without subscriptions (active or ended) when their historic
+  subscriptions have been deleted via the historic deletion worker. This means
+  we should be able to delete all subscribers (under a year old too) that don't
+  have any subscriptions (active or ended). However, we should be cautious
+  about doing this so we don't accidently remove new subscribers that lack
+  subscriptions, due to flaws in our code.  Some examples of where it was
+  unintentionally possible to end up with subscribers with no subscriptions can
+  be found [here][sub bug] and [here][sub bug two].
+
+[can't currently be made active again]: https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md#subscriber-and-subscription
+[seven day limit]: https://github.com/alphagov/email-alert-api/blob/f0c61539f7244bdb52db54d277a46412b728d642/app/services/auth_token_generator_service.rb#L7
+
+## Status
+
+Accepted
+
+## Consequences
+
+### Historic deletion worker
+
+We implemented a [worker] which runs periodically every day at [midday].
+Removing this data was deemed as non urgent and so deleting once a day, in
+hours, seemed sufficient.
+
+### The first historic deletion run
+
+On the first run of the worker which handles deleting historical data we
+removed over 33 million records. A breakdown of the amount of deletions per
+model can be found below:
+
+* ContentChange - 148,617
+
+* MatchedContentChange - 1,825,983
+
+* Message - 15
+
+* MatchedMessage - 16,099
+
+* DigestRun - 677
+
+* DigestRunSubscriber - 30,858,770
+
+* Subscription - 1,046,576
+
+* SubscriberList - 11,470
+
+* Subscriber - 196,376
+
+### Restrict analytics to 1 year of unused data
+
+We now restrict our unused data for analytics to a period of at most 1 year.
+This means any queries looking for this data (e.g. ended subscription's over a
+year old) will be met with no results. We don't expect this to be an issue due
+to our current expectation of the data we'll need for future analytics.
+
+### Modify the nullify process and retire the deactivated concept
+
+When documenting our data retention policy as described in this ADR, we found
+it difficult to understand and document the tangential concept of nullifying /
```

Comment:
We could also link to [the original explanation](https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-003-data-retention.md#subscriber-and-subscription) here.